packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn/AdyenDropIn.tsx

```diff
@@ -45,8 +50,29 @@ export const AdyenDropIn = memo<AdyenDropInProps>(() => {
     skip: true,
   });
 
+  const [adyenCheckoutSubmitParams, setAdyenCheckoutSubmitParams] = useState<{
+    state: AdyenCheckoutInstanceState;
+    component: DropinElement;
+  } | null>(null);
+
   const onSubmit: AdyenCheckoutInstanceOnSubmit = useEvent(async (state, component) => {
     component.setStatus("loading");
+    validateAllForms();
+    setAdyenCheckoutSubmitParams({ state, component });
+  });
+
+  const afterSubmit = useCallback(async () => {
+    if (!validating && !allFormsValid && adyenCheckoutSubmitParams) {
+      // validated, failed, let's reset the state
+      adyenCheckoutSubmitParams.component.setStatus("ready");
+      setAdyenCheckoutSubmitParams(null);
```

Comment:
Why do we want to clear the Adyen component data after validation fails?

Reply:
It doesn't clear the form, it just removes the reference to the adyen instance in the state. When you click on the Pay button again, the state will be set again and will wait until validation finishes… and if it's successful this time, Adyen will continue.